src/QsCompiler/CompilationManager/TypeChecking.cs

```diff
@@ -2207,6 +2217,15 @@ internal static void UpdateTypeChecking(this FileContentManager file, Compilatio
                     : contentTokens);
 
                 diagnostics = QsCompilerError.RaiseOnFailure(() => RunTypeChecking(compilation, declarationTrees, CancellationToken.None), "error on running type checking");
+
+                var numRemovedCycleDiags = file.CurrentSemanticDiagnostics().Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code())
+                    - diagnostics.Count(m => m.Code == ErrorCode.InvalidCyclicTypeParameterResolution.Code());
```

Comment:
Is it possible that some cycle diagnostics are removed and some are added since the last check, such that the difference is 0 and global type checking is not triggered?

Reply:
I'm not able to produce this situation. You could toggle between a local invalid cycle and a separate global invalid cycle with a single character change, but even then the behavior is what I'd expect. The errors correctly switch between the local cycle and the global cycle.

Reply:
I will admit this seems like a bit of a hacky way to check for this though. I'd appreciate any ideas you might have on how to make this check more robust.

Reply:
> You could toggle between a local invalid cycle and a separate global invalid cycle with a single character change

I think the language server can receive changes longer than one character (e.g. with copy and paste).

Reply:
Sure, but the effect is still the same: toggling between a local invalid cycle and a global invalid cycle.

Reply:
Actually now I seem to be triggering an issue sometimes. It seems to be somewhat random if the errors get cleared or not.

Reply:
fixed